src/languageFacts/entry.ts

```diff
@@ -4,33 +4,49 @@
  *--------------------------------------------------------------------------------------------*/
 'use strict';
 
-import { EntryStatus, IPropertyData, IAtDirectiveData, IPseudoClassData, IPseudoElementData, IValueData, MarkupContent, MarkupKind, MarkedString, HoverSettings } from '../cssLanguageTypes';
-
-export interface Browsers {
-	E?: string;
-	FF?: string;
-	IE?: string;
-	O?: string;
-	C?: string;
-	S?: string;
-	count: number;
-	all: boolean;
-	onCodeComplete: boolean;
-}
+import { EntryStatus, BaselineStatus, IPropertyData, IAtDirectiveData, IPseudoClassData, IPseudoElementData, IValueData, MarkupContent, MarkupKind, MarkedString, HoverSettings } from '../cssLanguageTypes';
 
 export const browserNames = {
-	E: 'Edge',
-	FF: 'Firefox',
-	S: 'Safari',
-	C: 'Chrome',
-	IE: 'IE',
-	O: 'Opera'
+	'C': {
+		name: 'Chrome',
+		platform: 'desktop'
+	},
+	'CA': {
+		name: 'Chrome',
+		platform: 'Android'
+	},
+	'E': {
+		name: 'Edge',
+		platform: 'desktop'
+	},
+	'FF': {
+		name: 'Firefox',
+		platform: 'desktop'
+	},
+	'FFA': {
+		name: 'Firefox',
+		platform: 'Android'
+	},
+	'S': {
+		name: 'Safari',
+		platform: 'macOS'
+	},
+	'SM': {
+		name: 'Safari',
+		platform: 'iOS'
+	}
 };
 
+const shortCompatPattern = /(E|FFA|FF|SM|S|CA|C|IE|O)([\d|\.]+)?/;
```

Comment:
@rviscomi Why are we looking for the `|` character in the `[\d|\.]` part? 🤔

I've analyzed the `browsers.css-data.json` file from the `@vscode/web-custom-data` package and haven't found a single property that has this character in the `browsers` or  `values.browsers` field.

But I've found 5 properties that have values in the `browsers` field, that don't match current pattern:

`overflow-anchor`: `'Spreview'`
`-moz-background-clip`: `'FF1-3.6'`
`-ms-filter`: `'IE8-9'`
`-ms-transform`: `'IE9-9'`
`-ms-transform-origin`: `'IE9-9'`

Actually, we are not interested in the version value (the second part of the regexp), only the browser name (the first part of the regexp).

Maybe we can do with a simpler regexp or without using a regexp at all?

Reply:
You can remove the `|`...